### DISCUSSION TITLE
Moving from jaeger+ES stack to jaeger+Tempo

### DISCUSSION THREAD

**Discussion by manojm321** [grafana/tempo#2782] [Q&A] [answered]
Hello,

Today we have jaeger-agent sending traces to jaeger-collector, which batches and sends traces to ElasticSearch. I would like to move to tempo for trace storage. Can I use  `grpc-plugin` in jaeger-collector to achieve this? what other alternatives do I have?

During the transition I'd like to write to both ES and Tempo.

**Comment by joe-elliott** [comment, marked as answer]
My recommendation would be to use the https://github.com/open-telemetry/opentelemetry-collector in your pipeline. It can consume traces and write to multiple targets. It supports exporting to all [Jaeger formats](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/jaegerexporter) and of course OTEL which Tempo supports.
